src/Context.h

```diff
@@ -115,6 +120,25 @@ class Context
   long                                time_sort_us        {0};
   long                                time_render_us      {0};
   long                                time_hooks_us       {0};
+
+  // the current task for DOM references, or NULL if there is no task
+  const Task *                        currentTask         {NULL};
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// CurrentTask resets Context::currentTask to NULL on destruction; this ensures
+// that this context value is not a dangling pointer.
+class CurrentTask {
+public:
+  ~CurrentTask();
+
+private:
+  CurrentTask(Context *context, const Task *previous);
+
+  Context *context;
```

Comment:
The context should never be `null`, and since it's a singleton we don't need to reassign it either during the lifetime of this object, what do you think about using a reference here instead?